src/compiler/checker.ts

```diff
@@ -18759,16 +18759,20 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     // An object type S is considered to be derived from an object type T if
     // S is a union type and every constituent of S is derived from T,
     // T is a union type and S is derived from at least one constituent of T, or
-    // S is a type variable with a base constraint that is derived from T,
+    // S is an intersection type and some constituent of S is derived from T, or
+    // S is a type variable with a base constraint that is derived from T, or
+    // T is {} and S is an object-like type (ensuring {} is less derived than Object), or

```

Comment:
q: what does "less derived" mean? how something can be more or less derived?

Reply:
It means that any object-like type, including `Object`, appears to be derived from `{}`. The effect of this addition is that `x instanceof Object` will narrow `{}` to `Object`, which in turn means that `x` qualifies as the right hand operand of an `in` check.

Reply:
I'd probably reword this like this then:
```suggestion
    // T is {} and S is an object-like type (ensuring that object-likes, including Object, are derived from {}), or
```

Maybe I'm just missing the appropriate background and lingo but this explanation is more readable to me.

Reply:
In the context of type theory, "less derived" is definitely a thing.  e.g. `Animal` is less derived than `Dog` which is less derived than `PitBull`, while `Cat` and `Dog` are equally derived, being sister types.